crates/nu-engine/src/print.rs

```diff
@@ -0,0 +1,34 @@
+use nu_errors::ShellError;
+use nu_source::Text;
+
+use crate::EvaluationContext;
+
+pub fn maybe_print_errors(context: &EvaluationContext, source: Text) -> bool {
+    let errors = context.current_errors.clone();
+    let mut errors = errors.lock();
+
+    if errors.len() > 0 {
+        let error = errors[0].clone();
+        *errors = vec![];
+
+        print_err(error, &source, context);
+        true
+    } else {
+        false
+    }
+}
+
+pub fn print_err(err: ShellError, source: &Text, ctx: &EvaluationContext) {
```

Comment:
We should try to create a clean separation between what is universal and what is specific to the platform.

Things like printing the the screen aren't going to be universal when you include wasm. The way I've been thinking about engine, is that it should be universal as much as possible. That would mean not putting printing here but further "out" from the core logic.

Reply:
I have added a function print_err to the host and moved this functionality there. We could provide a special implementation of the host for wasm builds in the future, so that errors are printed a little nicer :)